.github/workflows/cleanup.yml

```diff
@@ -1,14 +1,38 @@
 name: cleanup
 
 on:
-  schedule:
-    - cron: '*/5 * * * *'
+  # Triggerred on every workflow run except on master
+  workflow_run:
+    workflows: [ "ci" ]
+    types:
+      - requested
+    branches-ignore: [ "master" ]
+  # Triggerred on every PR close
+  pull_request:
+    types: [ "closed" ]
 
 jobs:
   cancel:
     runs-on: ubuntu-latest
+    timeout-minutes: 5
     steps:
-      - uses: n1hility/cancel-previous-runs@2e3c1893986568a2197c41957b9c809cbcf1a61e
+      - name: 'Cancel All Runs Except Latest'
+        # For pull_request events different config is needed
+        if: ${{ github.event_name != 'pull_request' }}
+        uses: styfle/cancel-workflow-action@0.9.0
         with:
-          token: ${{ github.token }}
-          workflow: ci.yml
+          # Cancel workflow runs scheduled after the current one. May help when queue is saturated.
+          # https://github.com/styfle/cancel-workflow-action#advanced-all-but-latest
+          all_but_latest: true
+          # Note: workflow_id can be a Workflow ID (number) or Workflow File Name (string). Here we have ID thanks to how this is triggered.
+          workflow_id: ${{ github.event.workflow.id }}
+          access_token: ${{ github.token }}
+      - name: 'Cancel Runs For Closed PRs'
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: styfle/cancel-workflow-action@0.9.0
+        with:
+          # Cancel workflow when PR closed. https://github.com/styfle/cancel-workflow-action#advanced-ignore-sha
+          ignore_sha: true
+          # Note: workflow_id can be a Workflow ID (number) or Workflow File Name (string). Here we don't have ID.
+          workflow_id: 1077735
```

Comment:
if the number stays (see  LO comment above), let's take a note where did the number come from

Reply:
Removed - filename works.